modules/misunderstood/src/migrations/v12_26_0-1630422173-migration_varchar_to_text.ts

```diff
@@ -0,0 +1,75 @@
+import * as sdk from 'botpress/sdk'
+import Knex from 'knex'
+
+import { FLAGGED_MESSAGE_STATUS, FLAGGED_MESSAGE_STATUSES, FLAG_REASON, RESOLUTION_TYPE } from '../types'
+
+const TABLE_NAME = 'misunderstood'
+const TEMP_TABLE_NAME = `${TABLE_NAME}_tmp`
+const COLUMN = 'preview'
+
+const migrateLite = async (db: sdk.KnexExtended, trx: Knex.Transaction, isUp) => {
+  //  1- create a temp table with the correct preview column type
+  const hastmp = await db.schema.transacting(trx).hasTable(TEMP_TABLE_NAME)
+  if (hastmp) {
+    await trx.raw(`TRUNCATE TABLE ${TEMP_TABLE_NAME}`)
+  } else {
+    await db.schema.transacting(trx).createTable(TEMP_TABLE_NAME, table => {
```

Comment:
is the else necessary there ?

Reply:
Yeah, we need the tmp table to exist

Reply:
Yeah, I know. Why not just truncate  if exist and create it directly ?
```
if (hasTmp){
    await trx.raw(`TRUNCATE TABLE ${TEMP_TABLE_NAME}`)
}
await db.schema.transacting(trx).createTable(TEMP_TABLE_NAME, table [...rest of your code...]
```

Reply:
Truncate doesn't delete the table, it just deletes all the data from it. If you try to create it after truncating, it should fail because the table already exists. It might be cleaner to do a proper delete though, I'll make that change.

Reply:
Yeah, good point. Thanks :)